### PR TITLE
👨‍👩‍👧 Use output of flow validation in mailroom to set flow dependencies

### PR DESCRIPTION
**Pull request by rowanseymour**

### CODE REVIEW

File: temba/flows/models.py [hunk 1]
```diff
@@ -2621,110 +2621,34 @@ def get_step_type(dest, rulesets, actionsets):
...
-        if len(fields):
-            existing = ContactField.user_fields.filter(org=self.org, key__in=fields).values_list("key")
+        # still need to do lazy creation of fields in the case of a flow import
+        if len(field_keys):
+            existing_keys = set(ContactField.user_fields.filter(org=self.org, key__in=field_keys).values_list("key"))
 
             # create any field that doesn't already exist
-            for field in fields:
-                if ContactField.is_valid_key(field) and field not in existing:
+            for key in field_keys:
```
Comment (rowanseymour) [inline temba/flows/models.py]: this shouldn't be here long term but I'd like to address that in another PR that reworks flow imports

**Comment by rowanseymour** [comment]
I checked on that there are the following tests which seem to cover all important bits:

* `FlowsTest.test_group_dependencies`
* `FlowsTest.test_flow_dependencies`
* `FlowsTest.test_update_dependencies_with_actiontype_flow`
* `FlowsTest.test_flow_delete_with_dependencies`

**Comment by nicpottier** [comment]
👍 We are missing channel dependencies in general but I suppose that should be done elsewhere.

**Comment by rowanseymour** [comment]
@nicpottier also message labels

**Pull request merged**